src/iconv.jl

```diff
@@ -1,9 +1,61 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
 module iconv
-import Base: close, eof, flush, read, readall, write
+import Base: close, eof, flush, read, readall, write, show
 import Base.Libc: errno, strerror, E2BIG, EINVAL, EILSEQ
 export StringEncoder, StringDecoder, encode, decode
+export StringEncodingError, OutputBufferError, IConvError
+export InvalidEncodingError, InvalidSequenceError, IncompleteSequenceError
+
+
+abstract StringEncodingError
+
+# specified encodings are not supported
+type InvalidEncodingError <: StringEncodingError
+    msg::UTF8String
```

Comment:
Sorry, this looks good, but I've come to think that instead of storing a `msg` field which would be identical for all instances; it would be better to have `message(::Type{InvalidEncodingError})`, etc., functions which will return the message template. Then it should be OK.

That should also allow you to get rid of inner constructors (or of most of them).